torch/hub.py

```diff
@@ -209,6 +212,58 @@ def _get_cache_or_reload(github, force_reload, verbose=True, skip_validation=Fal
 
     return repo_dir
 
+def _get_trusted_input(repo):
+    # separate function for testing
+    response = input(
+        f"The repository {repo} does not belong to the list of trusted repositories and as such cannot be downloaded. "
+        "Do you trust this repository and wish to add it to the trusted list of repositories (y/N)?")
+    return response
+
+def _check_repo(repo, trust_repo=None, calling_fn="load"):
+    hub_dir = get_dir()
+    filepath = os.path.join(hub_dir, "trusted_list")
+
+    if not os.path.exists(filepath):
+        Path(filepath).touch()
+
+        # Initialize with all existing repos
+        repos = next(os.walk(hub_dir))[1]
+        with open(filepath, "a") as file:
+            for _repo in repos:
+                file.write(_repo + "\n")
```

Comment:
I think this will add repos of the form `repoowner_reponame_branch` like `pytorch_vision_main`, and these won't be matched against what we actually check which is just `repoowner_reponame`.

Reply:
good point, that comes from the previous check with the `owner_repo_banch` strings.
We could split after the last `_` but that won't work when the branch is `my_branch_name`.
Do we keep track of what the owner / repo / branch is for a given cached repo?

Reply:
I don't think we do
What we can do is this: We have 3 sources for the checks 
(1) `_TRUSTED_REPO_OWNERS` tuple that checks the owners only
(2) `trusted_list` file, that checks `owner_repo`
(3) `trusted_list_legacy`  file, that checks `owner_repo_branch` since it's all we have